### PR TITLE
Added the ability to disable the dashboard

### LINKED ISSUE #1317: Make Kubernetes Dashboard optional
Currently Dashboard is not optional.  Add cluster.yaml variable to disable dashboard.

### PR DESCRIPTION
**Pull request by jasonrichardsmith**
templates: add boolean to make dashboard optional

Added disable flag to kubernetesDashboard to make the deployment optional

Fixes #1317

### CODE REVIEW

File: core/controlplane/config/templates/cloud-config-controller [hunk 1]
```diff
@@ -959,13 +961,13 @@ write_files:
 {{ end }}
 
       # Deployments
-      applyall "${mfdir}"/{kube-dns,kube-dns-autoscaler,kubernetes-dashboard,{{ if .Addons.ClusterAutoscaler.Enabled }}cluster-autoscaler,{{ end }}heapster{{ if .KubeResourcesAutosave.Enabled }},kube-resources-autosave{{ end }}}"-de.yaml"
```
[the remark this answers is not in the document]
Comment (c-knowles) [reply core/controlplane/config/templates/cloud-config-controller]: Yeah why not? But for items in the future is fine.

[the remark this answers is not in the document]
Comment (mumoshu) [reply core/controlplane/config/templates/cloud-config-controller]: What you mean by "items in the future"?

Anyway, I'm fine to make every single-lined `applyall` calls to multi-line as I said. There's no reason to not do that!

[the remark this answers is not in the document]
Comment (jasonrichardsmith) [reply core/controlplane/config/templates/cloud-config-controller]: Yes I will change disable to enable

**Comment by jasonrichardsmith** [comment]
I have made the requested change @mumoshu

**Comment by mumoshu** [comment]
@jasonrichardsmith Thank you very much for your effort!

**Comment by jasonrichardsmith** [comment]
@mumoshu thank you for all your work on this project.

File: core/controlplane/config/templates/cloud-config-controller (hunk 1 above)
[the remark this answers is not in the document]
Comment (c-knowles) [reply core/controlplane/config/templates/cloud-config-controller]: Turns out the space in this line causes install-kube-system to fail. Fixed in https://github.com/kubernetes-incubator/kube-aws/pull/1335.

**Pull request merged**